
Ported jack audio out to latest bio2jack library.

  - Rene Rebe <rene@exactcode.de>

--- MPlayer-1.0pre5/libao2/ao_jack.c.vanilla	2004-11-18 15:20:00.214822384 +0100
+++ MPlayer-1.0pre5/libao2/ao_jack.c	2004-11-18 15:22:29.162178944 +0100
@@ -34,8 +34,9 @@
 long JACK_Write(int deviceID, char *data, unsigned long bytes); /* returns the number of bytes written */
 long JACK_GetJackLatency(int deviceID); /* return the latency in milliseconds of jack */
 int  JACK_SetState(int deviceID, enum status_enum state); /* playing, paused, stopped */
-int  JACK_SetVolume(int deviceID, int left, int right); /* returns 0 on success */
-void JACK_GetVolume(int deviceID, int *left, int *right);
+int  JACK_SetVolumeForChannel(int deviceID, unsigned int channel, unsigned int volume);
+void JACK_GetVolumeForChannel(int deviceID, unsigned int channel, unsigned int *volume);
+
 //
 
 
@@ -59,7 +60,8 @@
 				ao_control_vol_t *vol = (ao_control_vol_t *)arg;
 				int l, r;
 				
-				JACK_GetVolume(driver, &l, &r);
+				JACK_GetVolumeForChannel(driver, 0, &l);
+				JACK_GetVolumeForChannel(driver, 1, &r);
 				vol->left = (float )l;
 				vol->right = (float )r;
 				
@@ -72,11 +74,16 @@
 					r = (int )vol->right,
 					err = 0;
 
-				if((err = JACK_SetVolume(driver, l, r))) {
+				if((err = JACK_SetVolumeForChannel(driver, 0, l))) {
 					mp_msg(MSGT_AO, MSGL_ERR, 
 							"AO: [Jack] Setting volume failed, error %d\n",err);
 					return CONTROL_ERROR;
 				}
+				if((err = JACK_SetVolumeForChannel(driver, 1, r))) {
+					mp_msg(MSGT_AO, MSGL_ERR,
+							"AO: [Jack] Setting volume failed, error %d\n",err);
+					return CONTROL_ERROR;
+				}
 				
 				return CONTROL_OK;
 			}
